lib/types/listen_request_body.ex

```diff
@@ -22,4 +23,21 @@ defmodule Kadena.Types.ListenRequestBody do
       {:error, _reasons} -> {:error, [listen: :invalid]}
     end
   end
+
+  defimpl JSONPayload do
+    alias Kadena.Types.ListenRequestBody
+    @type url :: String.t()
+    @type base_64_url :: Base64Url.t()
+
+    @impl true
+    def parse(%ListenRequestBody{listen: listen}) do
+      listen = extract_listen(listen)
+
+      %{listen: listen}
+      |> Jason.encode!()
```

Comment:
Just a minor change, this pipe is not necessary because there is just one function, so IMO it is much better to write:
`Jason.encode!(%{listen: listen})`